nntrainer/dataset/random_data_producers.cpp

```diff
@@ -0,0 +1,167 @@
+
+namespace nntrainer {
+
+/**
+ * @brief Props containing min value
+ *
+ */
+class PropsMin : public Property<float> {
+public:
+  /**
+   * @brief Construct a new props min object with a default value
+   *
+   * @param value default value
+   */
+  PropsMin(float value = 0.0f) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key = "min"; /**< unique key to access */
+  using prop_tag = float_prop_tag;          /**< property type */
+};
+
+/**
+ * @brief Props containing max value
+ *
+ */
+class PropsMax : public Property<float> {
+public:
+  /**
+   * @brief Construct a new props max object with a default value
+   *
+   * @param value default value
+   */
+  PropsMax(float value = 1.0f) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key = "max"; /**< unique key to access */
+  using prop_tag = float_prop_tag;          /**< property type */
+};
+
+/**
+ * @brief Props containing number of iterations
+ *
+ */
+class PropsDataSize : public Property<unsigned int> {
+public:
+  /**
+   * @brief Construct a new props data size object with a default value
+   *
+   * @param value default value
+   */
+  PropsDataSize(unsigned int value = 512) :
+    nntrainer::Property<unsigned int>(value) {}
+  static constexpr const char *key = "size"; /**< unique key to access */
+  using prop_tag = uint_prop_tag;            /**< property type */
+};
+
+RandomDataOneHotProducer::RandomDataOneHotProducer() :
+  rd_one_hot_props(new Props()) {}
+
+RandomDataOneHotProducer::~RandomDataOneHotProducer() {}
+
+const std::string RandomDataOneHotProducer::getType() const {
+  return RandomDataOneHotProducer::type;
+}
+
+unsigned long long
+RandomDataOneHotProducer::size(const std::vector<TensorDim> &input,
+                               const std::vector<TensorDim> &output) const {
+  return std::get<PropsDataSize>(*rd_one_hot_props).get();
+}
+
+void RandomDataOneHotProducer::setProperty(
+  const std::vector<std::string> &properties) {
+  auto left = loadProperties(properties, *rd_one_hot_props);
+  NNTR_THROW_IF(!left.empty(), std::invalid_argument)
+    << "There are unparsed properties, size: " << left.size();
+}
+
+DataProducer::Gernerator
+RandomDataOneHotProducer::finalize(const std::vector<TensorDim> &input_dims,
+                                   const std::vector<TensorDim> &label_dims) {
+  /** check if the given producer is ready to finalize */
+  auto &[min_, max_, sz] = *rd_one_hot_props;
+
+  NNTR_THROW_IF(std::any_of(label_dims.begin(), label_dims.end(),
+                            [](const TensorDim &dim) {
+                              return dim.channel() != 1 || dim.height() != 1;
+                            }),
+                std::invalid_argument)
+    << "Label dimension containing channel or height not allowed";
+
+  NNTR_THROW_IF(min_.get() > max_.get(), std::invalid_argument)
+    << "Min value is bigger then max value, min: " << min_.get()
+    << "max: " << max_.get();
```

Comment:
So we just check the min, max values during finalize, not during set property.

Reply:
Yes :), setProperty validates if given property is  valid. finalize validates if given combination is valid.